### PR TITLE
Added domain resource support

### LINKED ISSUE #11: Add domain resource support
Add support for `domain` resources.

- [ ] Create `Domain` and `DomainRequest` structs in `domain.go`.
- [ ] Add `ListDomains`, `GetDomain`, and `CreateDomain` methods.

See lines [370](https://github.com/xaptum/enfcli/blob/1a4b867c4c59cd99214f5b774f20f42855062967/lib/enfapi.rb#L370), [376](https://github.com/xaptum/enfcli/blob/1a4b867c4c59cd99214f5b774f20f42855062967/lib/enfapi.rb#L376), and [382](https://github.com/xaptum/enfcli/blob/1a4b867c4c59cd99214f5b774f20f42855062967/lib/enfapi.rb#L382) of the `enfcli` for the API endpoints.

Only "Xaptum Admin" users can list and create domains.  @venkat-xaptum can give you the "Xaptum Admin" username/password for our dev infrastructure.

### PR DESCRIPTION
**Pull request by dylanirlbeck**
Fixes #11 

I also made a small change to the `.gitignore`

### CODE REVIEW

File: enf/domain.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	return body.(*domainResponse).Data[0], resp, nil
+}
+
+// ActivateDomain activates the given domain (sets the status field to ACTIVE)
+func (s *DomainService) ActivateDomain(ctx context.Context, domain string) (*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains/%v/status", domain)
+	// The ActivateDomain endpoint will accept any request body, so there's no need to have a *DomainRequest parameter.
+	// Instead, just use an empty DomainRequest struct as the "body".
+	body, resp, err := s.client.put(ctx, path, new(domainResponse), DomainRequest{})
```
Comment (drbild) [inline enf/domain.go]: `DomainRequest` isn't related to this API call, so using an empty one as the "body" feels weird.  How about an [empty struct](https://medium.com/@l.peppoloni/how-to-improve-your-go-code-with-empty-structs-3bd0c66bc531) instead?

```
body, resp, err := s.client.put(ctx, path, new(domainResponse), struct{}{})
```

Comment (dylanirlbeck) [reply enf/domain.go]: I didn't even consider this, but you're totally right. An empty struct is more natural.

File: enf/domain.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		return nil, resp, err
+	}
+
+	return body.(*domainResponse).Data[0], resp, nil
+}
+
+// ActivateDomain activates the given domain (sets the status field to ACTIVE)
+func (s *DomainService) ActivateDomain(ctx context.Context, domain string) (*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains/%v/status", domain)
+	// The ActivateDomain endpoint will accept any request body, so there's no need to have a *DomainRequest parameter.
```
Comment (drbild) [inline enf/domain.go]: Maybe remove this comment too?

Comment (dylanirlbeck) [reply enf/domain.go]: Ah, yes I forgot about the comments. I'll remove those and then merge.

**Pull request merged**